src/main/scala/viper/server/ViperCoreServerUtils.scala

```diff
@@ -0,0 +1,89 @@
+package viper.server
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.stream.scaladsl.{Keep, Sink, Source, SourceQueueWithComplete}
+import akka.pattern.ask
+import akka.util.Timeout
+import viper.silver.reporter.{EntityFailureMessage, Message, OverallFailureMessage, OverallSuccessMessage}
+import viper.silver.verifier.{AbstractError, VerificationError, VerificationResult, Failure => VerificationFailure, Success => VerificationSuccess}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success}
+
+object ViperCoreServerUtils{
+  implicit private val executionContext = ExecutionContext.global
+
+  private object SeqActor {
+    case object Result
+    def props(): Props = Props(new SeqActor())
+  }
+
+  class SeqActor() extends Actor{
+
+    var messages: List[Message] = List()
+    var messages_promise: Promise[List[Message]] = Promise[List[Message]]()
+
+    override def receive: PartialFunction[Any, Unit] = {
+      case m: Message =>
+        messages = messages :+ m
+        m match {
+          //once the overall messages are last to arrive and indicate that the promise can be finalized with success.
+          case _: OverallSuccessMessage =>
+            messages_promise success messages
+          case _: OverallFailureMessage =>
+            messages_promise success messages
+          case _ =>
+        }
+      case SeqActor.Result =>
+        messages_promise.future
+        sender() ! messages_promise
+      case e: Throwable =>
+        //receiving an error means the promise can be finalized with failure.
+        messages_promise failure e
+    }
+  }
+
+  /** Get a Future containing all messages generated by the backend.
+    *
+    * This is a utility function and not part of ViperCoreServer. Therefore, an instance of ViperCoreServer as well as
+    * an instance of an actor system must be provided.
+    *
+    * Deletes the jobhandle on completion.
+    */
+  def getMessagesFuture(core: ViperCoreServer, jid: Int, actor_system: ActorSystem): Future[List[Message]] = {
+    val actor = actor_system.actorOf(SeqActor.props())
+    core.streamMessages(jid, actor)
+    implicit val askTimeout: Timeout = Timeout(core.config.actorCommunicationTimeout() milliseconds)
```

Comment:
Unless ```actorCommunicationTimeout``` is a factory, you should probably write ```core.config.actorCommunicationTimeout.milliseconds``` without the parents. 

This is the common way to call getters in Scala.

Reply:
`core.config.actorCommunicationTimeout` is a scallopOption[Int]. 
`actorCommunicationTimeout()` returns that integer.
`core.config.actorCommunicationTimeout.milliseconds` is syntactically wrong.